vajram/extensions/graphql/vajram-graphql-codegen/src/main/java/com/flipkart/krystal/vajram/graphql/codegen/SchemaReaderUtil.java

```diff
@@ -473,4 +473,40 @@ public Optional<GraphQLTypeName> getComposingEntityType(ObjectTypeDefinition typ
                 typeDefinition, Directives.COMPOSED_TYPE, DirectiveArgs.IN_ENTITY)
             .map(GraphQLTypeName::of);
   }
+
+  /** Gets the Java type mapping for a scalar type from the @javaType directive. */
+  public Optional<ClassName> getJavaTypeForScalar(String scalarName) {
+    return typeDefinitionRegistry
+        .getType(scalarName, ScalarTypeDefinition.class)
+        .flatMap(
+            scalarDef -> {
+              List<Directive> javaTypeDirectives = scalarDef.getDirectives(JAVA_TYPE_DIRECTIVE);
+              if (javaTypeDirectives.isEmpty()) {
+                return Optional.empty();
+              }
+
+              Directive directive = javaTypeDirectives.getFirst();
```

Comment:
This code is being repeated in two places. Move to a utility method which accepts a DirectivesContainer and returns a className